app-backend/db/src/test/scala/com/azavea/rf/database/ProjectDaoSpec.scala

```diff
@@ -274,4 +274,197 @@ class ProjectDaoSpec extends FunSuite with Matchers with Checkers with DBTestCon
       }
     }
   }
+  // addPermission and getPermissions
+  test("add a permission to a project and get it back") {
+    check {
+      forAll {
+        ( userTeamOrgPlat: (User.Create, Team.Create, Organization.Create, Platform),
+          acr: ObjectAccessControlRule,
+          project: Project.Create,
+          grantedUser: User.Create
+        ) => {
+          val projectPermissionIO = for {
+            projMiscInsert <- fixUpProjMiscInsert(userTeamOrgPlat, project)
+            (projectInsert, dbUserTeamOrgPlat) = projMiscInsert
+            dbGrantedUser <- UserDao.create(grantedUser)
+            dbUserGrantedTeamOrgPlat = dbUserTeamOrgPlat.copy(_1 = dbGrantedUser)
+            acrInsert = fixUpObjectAcr(acr, dbUserGrantedTeamOrgPlat)
+            perms <- ProjectDao.addPermission(projectInsert.id, acrInsert)
```

Comment:
For any of these values that you don't end up needing to use, you can throw them away with

```scala
_ <- FooDao.doAThing(...)
```

which makes it clearer that we only care about the side-effect.